server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -212,6 +212,23 @@ func (e *EventChannel) HandleEvent(ctx context.Context, event *pepb.PublishBuild
 		}
 	}
 
+	// When we get the workspace status event, update the invocation in the DB
+	// so that it can be searched by its commit SHA, user name, etc. even
+	// while the invocation is still in progress.
+	if isWorkspaceStatusEvent(bazelBuildEvent) {
```

Comment:
*generally* you'd want to make sure the read and write were in the same transaction so you didn't accidentally get some funky state happening between read and write -- but in this case I think it's probably "ok" to do this since all invocation-modifying events are coming through the PublishBuildEvent stream, and you won't process the next event until this one is handled.

Can you add a TODO to put these in a transaction at some point? It's low-priority but a reminder to us if we encounter weirdness around this later down the line.

Reply:
Done